src/compiler/checker.ts

```diff
@@ -24004,6 +24004,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     // we perform type inference (i.e. a type parameter of a generic function). We cache
     // results for union and intersection types for performance reasons.
     function couldContainTypeVariables(type: Type): boolean {
+        if (type.flags & TypeFlags.Intrinsic) return false;

```

Comment:
I'm not understanding how the function would previously return `true` for the wildcard type (which is a `TypeFlags.Any`)? What am I missing?

Reply:
The place I was referring to was inferTypes, which starts with a wildcard check, but if both sides are a wildcard we loop forever.

But, it checks for "could contain type var" before the wildcard logic.

Reply:
Right, but assuming both sides are wildcard types, how is it that we loop forever? That would require `couldContainTypeVariables` to return `true` for a `TypeFlags.Any`, and I'm not seeing how that happens.

Reply:
função poderiaContainTypeVariables e sua manipulação de tipos curinga (TypeFlags.Any) mas é necessária mais depuração para entender a estranha corrupção.

Reply:
Just to note the scary thing I'm seeing:

![image](https://github.com/microsoft/TypeScript/assets/5341706/9686670d-f94f-4aae-a8b4-d1add8e6feca)

Reply:
Traced this down to the new code in #53246, but it's absolutely bizarre. The flag is clearly there that says that "could contain type vars" is computed, and yet it still proceeds to overwrite it for no reason.